### PR TITLE
Update sample to use MSAL 0.4.0 

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: Graph-iOS-Swift-Snippets/AppDelegate.swift [hunk 1]
```diff
@@ -6,37 +6,13 @@
...
-class AppDelegate: UIResponder, UIApplicationDelegate {
-
+class AppDelegate: UIResponder, UIApplicationDelegate
+{
     var window: UIWindow?
 
-
-    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
-        // Override point for customization after application launch.
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool
```
Comment (oldalton) [inline Graph-iOS-Swift-Snippets/AppDelegate.swift:14]: You need to make sure to add MSAL handleResponse redirect in here.

File: Graph-iOS-Swift-Snippets/AuthenticationProvider.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    func acquireAuthToken(scopes: [String], completion:@escaping (_ success:Bool, _ error: NSError?) -> Void)
+    {
+        var completionBlock: MSALCompletionBlock?
+        completionBlock = { (result, error) in
+            
+            if let error = error as NSError?
+            {
+                if (error.domain == MSALErrorDomain)
+                {
+                    let msalError = MSALError(rawValue: error.code)!
```
Comment (oldalton) [inline Graph-iOS-Swift-Snippets/AuthenticationProvider.swift:31]: why do we need to create MSALError?

File: Graph-iOS-Swift-Snippets/AuthenticationProvider.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                        var accounts = [MSALAccount]()
+                        do {
+                            accounts = try self.msalClient.allAccounts()
+                        } catch let error as NSError {
+                            print("Error: ", error)
+                            completion(false, error)
+                            return
+                        }
+                        
+                        if let account = accounts.first {
```
Comment (oldalton) [inline Graph-iOS-Swift-Snippets/AuthenticationProvider.swift]: should we have a separate function to call acquire token silent with parameters? Would make this error handling a bit more readable.

File: Graph-iOS-Swift-Snippets/Info.plist [hunk 4]
```diff
@@ -46,5 +46,23 @@
...
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>msauth.com.microsoft.Graph-iOS-Swift-Snippets</string>
+				<string>auth</string>
+			</array>
+		</dict>
+	</array>
+	<key>LSApplicationQueriesSchemes</key>
+	<array>
+		<string>msauthv2</string>
```
Comment (oldalton) [inline Graph-iOS-Swift-Snippets/Info.plist:65]: please also add msauth

**Review by oldalton — CHANGES_REQUESTED** [review]

File: Graph-iOS-Swift-Snippets/AuthenticationProvider.swift (hunk 2 above)
Comment (antrix1989) [reply Graph-iOS-Swift-Snippets/AuthenticationProvider.swift:31]: We use it in switch below. We just map integer to enum here.

File: Graph-iOS-Swift-Snippets/Info.plist (hunk 4 above)
Comment (antrix1989) [reply Graph-iOS-Swift-Snippets/Info.plist:65]: done

File: Graph-iOS-Swift-Snippets/AppDelegate.swift (hunk 1 above)
Comment (antrix1989) [reply Graph-iOS-Swift-Snippets/AppDelegate.swift:14]: done

File: Graph-iOS-Swift-Snippets/AuthenticationProvider.swift (hunk 3 above)
Comment (antrix1989) [reply Graph-iOS-Swift-Snippets/AuthenticationProvider.swift]: done

File: Graph-iOS-Swift-Snippets/ApplicationConstants.swift [hunk 5]
```diff
@@ -6,28 +6,31 @@
 
 import Foundation
 
-
-struct ApplicationConstants {
+struct ApplicationConstants
+{
     // You will set your application's clientId
-    static let clientId = "619a3897-5a78-4a84-b5a8-4bcd22100f29"
+    static let clientId = "YOUR CLIENT ID"
```
Comment (oldalton) [inline Graph-iOS-Swift-Snippets/ApplicationConstants.swift:12]: Should we add some client ID here so sample works?

**Review by oldalton — APPROVED** [review]

File: Graph-iOS-Swift-Snippets/ApplicationConstants.swift (hunk 5 above)
Comment (Jackson-Woods) [reply Graph-iOS-Swift-Snippets/ApplicationConstants.swift:12]: In general, no. We try to be consistent across Graph samples - the instructions explain how to obtain a client id.

**Pull request merged**